spring-cloud-stream-binders/spring-cloud-stream-binder-rabbit/src/main/java/org/springframework/cloud/stream/binder/rabbit/RabbitMessageChannelBinder.java

```diff
@@ -291,70 +291,63 @@ public void onInit() {
 	private Binding<MessageChannel> doRegisterConsumer(final String name, String group, MessageChannel moduleInputChannel, Queue queue,
 													   final RabbitConsumerProperties properties) {
 		DefaultBinding<MessageChannel> consumerBinding = null;
-		// TODO https://github.com/spring-cloud/spring-cloud-stream/issues/401
-		ClassLoader originalClassloader = Thread.currentThread().getContextClassLoader();
-		try {
-			ClassUtils.overrideThreadContextClassLoader(SimpleMessageListenerContainer.class.getClassLoader());
-			SimpleMessageListenerContainer listenerContainer = new SimpleMessageListenerContainer(
-					this.connectionFactory);
-			listenerContainer.setAcknowledgeMode(properties.getAcknowledgeMode());
-			listenerContainer.setChannelTransacted(properties.isTransacted());
-			listenerContainer.setDefaultRequeueRejected(properties.isRequeueRejected());
-
-			int concurrency = properties.getConcurrency();
-			concurrency = concurrency > 0 ? concurrency : 1;
-			listenerContainer.setConcurrentConsumers(concurrency);
-			int maxConcurrency = properties.getMaxConcurrency();
-			if (maxConcurrency > concurrency) {
-				listenerContainer.setMaxConcurrentConsumers(maxConcurrency);
-			}
-
-			listenerContainer.setPrefetchCount(properties.getPrefetch());
-			listenerContainer.setTxSize(properties.getTxSize());
-			listenerContainer.setTaskExecutor(new SimpleAsyncTaskExecutor(queue.getName() + "-"));
-			listenerContainer.setQueues(queue);
-			int maxAttempts = properties.getMaxAttempts();
-			if (maxAttempts > 1 || properties.isRepublishToDlq()) {
-				RetryOperationsInterceptor retryInterceptor = RetryInterceptorBuilder.stateless()
-						.maxAttempts(maxAttempts)
-						.backOffOptions(properties.getBackOffInitialInterval(),
-								properties.getBackOffMultiplier(),
-								properties.getBackOffMaxInterval())
-						.recoverer(determineRecoverer(name, properties.getPrefix(), properties.isRepublishToDlq()))
-						.build();
-				listenerContainer.setAdviceChain(new Advice[] { retryInterceptor });
+		ClassUtils.overrideThreadContextClassLoader(SimpleMessageListenerContainer.class.getClassLoader());
```

Comment:
We can do away with overriding the TCCL here, too, I think, especially as it is not restored at the end of the method. This has little relevance for Boot applications anyway.